k8scrdclient/k8scrdclient.go

```diff
@@ -153,3 +157,90 @@ func (c *CRDClient) ensureStatusSubresourceCreated(ctx context.Context, crd *api
 
 	return nil
 }
+
+// crdVersionLatest returns true when the desired version of the given CRD
+// represents the latest version of them all. In case the current version of the
+// given CRD which we just read from the system appears to be newer, this means
+// another version of the same operator updated the CRD to its own version,
+// because it was in fact the latest version. Consider the following
+// relationships between operators and CRD versions they are aware of. Below o1
+// is operator-1 being aware of apiversion v1. In turn operator-2 knows about
+// apiversion v1 and v2, where v1 must be the exact same version as operator-1
+// defined already in order to keep the CRD schema version lifecycle in tact.
+//
+//         o1    |     o2     |    o3
+//     ---------------------------------
+//         v1    |   v1, v2   |   v2, v3
```

Comment:
I do not get this. From the previous explanation I would have expected each operator to only "know" about exactly one crd version. It then compares that one version to the latest version it can get from the api server. If it's not latest then it does nothing.

Reply:
Discussed with Marcel. I improved the comment to make more clear that o1 is not running anymore when o3 is deployed.